File: 119. Pascal's Triangle II.js
/**
 * @param {number} rowIndex
 * @return {number[]}
 */
var getRow = function(rowIndex) {
    if (rowIndex === 0)
        return [1];
    else if (rowIndex == 1)
        return [1,1];
    let res = [1,1];
    for (let i = 2; i <= rowIndex; i++)
    {
        let temp = [1];
        for (let j = 1; j < res.length; j++)
            temp.push(res[j-1] + res[j]);
        res = temp.concat([1]);
    }
    return res;
};

File: 119. Pascal's Triangle II.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        if rowIndex == 0:
            return [1]
        elif rowIndex == 1:
            return [1,1]
        res = [1,1]
        for i in xrange(2, rowIndex+1):
            res = [1] + [res[i-1] + res[i] for i in xrange(1, len(res))] +[1]
        return res
